core/platform/auth/firebase_auth_services_test.py

```diff
@@ -635,69 +665,71 @@ def swap_delete_user_to_always_fail(self):
 
-    def test_delete_auth_associations_retry_after_unverified_successful_attempt(
+    def test_verify_delete_external_auth_associations_after_failed_attempt(
             self):
-        self.delete_auth_associations()
+        self.delete_external_auth_associations()
 
         self.assert_firebase_account_is_deleted()
 
         with self.swap_initialize_sdk_to_always_fail():
-            self.assert_auth_associations_are_not_deleted()
+            self.assert_external_auth_associations_are_not_deleted()
 
-        self.delete_auth_associations()
+        self.delete_external_auth_associations()
 
-        self.assert_auth_associations_are_deleted()
+        self.assert_external_auth_associations_are_deleted()
 
-    def test_delete_auth_associations_when_delete_user_fails(self):
+    def test_delete_external_auth_associations_when_delete_user_fails(self):
         with self.swap_delete_user_to_always_fail():
-            self.delete_auth_associations()
+            self.delete_external_auth_associations()
 
         self.assert_firebase_account_is_not_deleted()
 
-        self.assert_auth_associations_are_not_deleted()
+        self.assert_external_auth_associations_are_not_deleted()
 
-    def test_delete_auth_associations_when_get_user_fails(self):
-        self.delete_auth_associations()
+    def test_delete_external_auth_associations_when_get_user_fails(self):
+        self.delete_external_auth_associations()
 
         self.assert_firebase_account_is_deleted()
 
         with self.swap_get_user_to_always_fail():
-            self.assert_auth_associations_are_not_deleted()
+            self.assert_external_auth_associations_are_not_deleted()
 
-        self.assert_auth_associations_are_deleted()
+        self.assert_external_auth_associations_are_deleted()
 
-    def test_delete_auth_associations_when_init_fails_during_delete(self):
+    def test_delete_external_auth_associations_when_init_fails_during_delete(
+            self):
         with self.swap_initialize_sdk_to_always_fail():
-            self.delete_auth_associations()
+            self.delete_external_auth_associations()
 
         self.assert_firebase_account_is_not_deleted()
 
-        self.assert_auth_associations_are_not_deleted()
+        self.assert_external_auth_associations_are_not_deleted()
 
-    def test_delete_auth_associations_when_init_fails_during_verify(self):
-        self.delete_auth_associations()
+    def test_delete_external_auth_associations_when_init_fails_during_verify(
+            self):
+        self.delete_external_auth_associations()
 
         self.assert_firebase_account_is_deleted()
 
         with self.swap_initialize_sdk_to_always_fail():
-            self.assert_auth_associations_are_not_deleted()
+            self.assert_external_auth_associations_are_not_deleted()
 
-        self.assert_auth_associations_are_deleted()
+        self.assert_external_auth_associations_are_deleted()
```

Comment:
> The auth_associations are read-only functions, but they require a Firebase connection to return a valid value. The swap tries to test connection failures, hence why the value changes depending on whether the SDK is being mocked to simulate failures.

I have two issues with this:

(a) Shouldn't connection failures result in an error and not True/False?

(b) Assertions should not depend on the test environment, so this is still confusing -- I would not expect, say, assertEqual() to behave differently inside and outside a `with` block. I suggest unpacking the assertion to make clear what exactly is failing -- in this case, presumably the call to firebase_auth_services.are_auth_associations_deleted().

Reply:
(a) I don't think so, a connection failure means we don't know for sure → we cannot say that the associations are deleted. I agree that an Exception makes more sense, but none of the code in wipeout_services uses errors to communicate errors so I was following that example. I'll defer to @vojtechjelinek's opinion for the best way to handle these kinds of issues.

(b) Hmm I'm not sure I follow your reasoning, we use swaps all the time to mock environment changes (e.g. mocking `datetime.datetime.utcnow()`), why wouldn't a connection failure fit with that pattern?

Reply:
Actually, scrap (b). I think the main issue that is causing confusion is (a).

I think assert_external_auth_associations_are_deleted() should raise an error, because it saying true or false is logically incorrect. Wipeout doesn't typically handle external stuff so I think the fact that existing functions don't use exceptions is not an indicator that they shouldn't be used in this case.

Reply:
Err, just to be clear, are we talking about the test method (`assert_external_auth_associations_are_deleted`) or the actual API (`are_external_auth_associations_deleted`)?

Reply:
I ended up really liking the name `verify_external_auth_associations_are_deleted`, so I changed it anyway.

WDYT?

Reply:
I really still think you should raise, conceptually. There are three outcomes of the verification check: either it is true, or it is not true, or the server shrugs and says "I don't know". These are conceptually separate and we should not be equating "I don't know" to "false" in the API that connects to the server, although code higher up might do so.

@vojtechjelinek, this might need your input. What do you think?

Reply:
I think the disconnect here is from the meaning I'm placing into "verify" as a verb.

This function returns false when we could not **_verify_** whether the external auth associations have been deleted.

There are only 2 scenarios here, 3 situations in total:
1. The server responds with "yes assocs are deleted" → True (known to be True → verify is True)
2. The server responds with "no assocs are deleted" → False (known to be False → verify is False)
3. The server could not be reached → False (status unknown → verify is False)

I don't see value in disambiguating 2 from 3, especially considering that the way verify is being defined here already correctly encapsulates both distinct options.

Reply:
Resolved offline, function is fine but extra comments have been added to highlight how it is used by `wipeout_service`.